app/src/main/java/org/sopt/soptandroidseminar/view/signin/SignInActivity.kt

```diff
@@ -64,12 +65,13 @@ class SignInActivity : AppCompatActivity() {
     }
 
     private fun observeSuccessLogin() {
-        viewModel.successLogIn.observe(this) {
-            goToHomeActivity()
-        }
-        viewModel.failureLogIn.observe(this) {
-            showToast("로그인에 실패했습니다.")
-        }
+        viewModel.logIn.observe(this, EventObserver{
+            when(it) {
+                SignInViewModel.LOGIN_SUCCESS -> goToHomeActivity()
+                SignInViewModel.LOGIN_FAILURE -> showToast("아이디/비밀번호 를 확인해주세요.")
+                SignInViewModel.SERVER_FAILURE -> showToast("서버통신에 실패했습니다.")
```

Comment:
```suggestion
                else -> showToast(it.message)
```

이 방식도 사실 보완해야할 점이 있긴 한데 실패인 경우에는 토스트를 띄우고싶다가 공통된 로직이라면 이런식으로도 enum을 활용할 수 있을 것 같아.